src/borg/platform/freebsd.pyx

```diff
@@ -50,59 +50,77 @@ cdef extern from "unistd.h":
 
+
+def listxattr(path, *, follow_symlinks=False):
     def func(path, buf, size):
         if isinstance(path, int):
-            return c_extattr_list_fd(path, ns, <char *> buf, size)
+            return c_extattr_list_fd(path, ns_id, <char *> buf, size)
         else:
             if follow_symlinks:
-                return c_extattr_list_file(path, ns, <char *> buf, size)
+                return c_extattr_list_file(path, ns_id, <char *> buf, size)
             else:
-                return c_extattr_list_link(path, ns, <char *> buf, size)
+                return c_extattr_list_link(path, ns_id, <char *> buf, size)
 
+    ns = b"user"
+    ns_id = NS_ID_MAP[ns]
     n, buf = _listxattr_inner(func, path)
-    return [prefix + name for name in split_lstring(buf[:n]) if name]
+    return [ns + b"." + name for name in split_lstring(buf[:n]) if name]
 
 
 def getxattr(path, name, *, follow_symlinks=False):
-    ns, prefix = EXTATTR_NAMESPACE_USER, b'user.'
-
     def func(path, name, buf, size):
         if isinstance(path, int):
-            return c_extattr_get_fd(path, ns, name, <char *> buf, size)
+            return c_extattr_get_fd(path, ns_id, name, <char *> buf, size)
         else:
             if follow_symlinks:
-                return c_extattr_get_file(path, ns, name, <char *> buf, size)
+                return c_extattr_get_file(path, ns_id, name, <char *> buf, size)
             else:
-                return c_extattr_get_link(path, ns, name, <char *> buf, size)
+                return c_extattr_get_link(path, ns_id, name, <char *> buf, size)
 
-    # strip namespace if there, but ignore if not there.
-    # older borg / attic versions did not prefix the namespace to the names.
-    if name.startswith(prefix):
-        name = name[len(prefix):]
+    ns, name = split_ns(name, b"user")
+    ns_id = NS_ID_MAP[ns]  # this will raise a KeyError it the namespace is unsupported
     n, buf = _getxattr_inner(func, path, name)
     return bytes(buf[:n])
 
 
 def setxattr(path, name, value, *, follow_symlinks=False):
-    ns, prefix = EXTATTR_NAMESPACE_USER, b'user.'
-
     def func(path, name, value, size):
         if isinstance(path, int):
-            return c_extattr_set_fd(path, ns, name, <char *> value, size)
+            return c_extattr_set_fd(path, ns_id, name, <char *> value, size)
         else:
             if follow_symlinks:
-                return c_extattr_set_file(path, ns, name, <char *> value, size)
+                return c_extattr_set_file(path, ns_id, name, <char *> value, size)
             else:
-                return c_extattr_set_link(path, ns, name, <char *> value, size)
+                return c_extattr_set_link(path, ns_id, name, <char *> value, size)
 
-    # strip namespace if there, but ignore if not there.
-    # older borg / attic versions did not prefix the namespace to the names.
-    if name.startswith(prefix):
-        name = name[len(prefix):]
-    _setxattr_inner(func, path, name, value)
```

Comment:
this was problematic in case an archive from another platform (e.g. with `name == b"security.selinux"`) was restored on freebsd.

it then used `EXTATTR_NAMESPACE_USER` and `name == b"security.selinux"` in the API call.

also, it doesn't make sense to restore selinux stuff onto freebsd anyway, right?